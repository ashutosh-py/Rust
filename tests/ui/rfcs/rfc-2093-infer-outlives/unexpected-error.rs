struct Foo<T> { t: &'static T }

fn main() {
    let x = 3;
    let _ = Foo { t: &x }; //~ ERROR `x` does not live long enough
}
