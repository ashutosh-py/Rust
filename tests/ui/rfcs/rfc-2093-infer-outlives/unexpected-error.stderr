error[E0597]: `x` does not live long enough
  --> $DIR/unexpected-error.rs:5:22
   |
LL |     let x = 3;
   |         - binding `x` declared here
LL |     let _ = Foo { t: &x };
   |                      ^^
   |                      |
   |                      borrowed value does not live long enough
   |                      this usage requires that `x` is borrowed for `'static`
LL | }
   | - `x` dropped here while still borrowed

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0597`.
